fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerIndexFromIndexTest.java

```diff
@@ -66,10 +68,23 @@ private void populateData(final long numRecords) {
     }
 
     private void buildSrcIndex(Index srcIndex) {
-        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
-                .setDatabase(fdb).setMetaData(metaData).setIndex(srcIndex).setSubspace(subspace)
-                .build()) {
-            indexer.buildIndex(true);
+        // temporary loop to prevent test failures on a trivial source index built.
+        int sanity = 5;
+        for (boolean built = false; !built;) {
+            try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                    .setDatabase(fdb).setMetaData(metaData).setIndex(srcIndex).setSubspace(subspace)
+                    .build()) {
+                indexer.buildIndex(true);
+                built = true;
+            } catch (Exception ex) {
+                if (-- sanity <= 0) {
+                    throw ex;
+                }
+                Logger logger = LoggerFactory.getLogger(OnlineIndexerIndexFromIndexTest.class);
+                if (logger.isWarnEnabled()) {
+                    logger.warn("Attempt failed" , ex);
+                }
+            }
```

Comment:
I think it is important to look into the underlying cause. Looking at: https://github.com/FoundationDB/fdb-record-layer/blob/b2d961a47ae3c7f160c2d3affa32c5f971ef2dcc/fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerTest.java#L83 the timeout for building an index is 5 minutes (!) and looking at the log lines from the test failure referenced in my comments in the issue:

```
2021-08-09 17:12:23,507 [DEBUG] c.a.f.r.p.f.FDBRecordStore - indexes need rebuilding indexesReadable="[MySimpleRecord$num_value_unique, MySimpleRecord$num_value_3_indexed, MySimpleRecord$str_value_indexed]" new_store="true" record_count="0" subspace="\x15("
2021-08-09 17:17:23,857 [WARN] c.a.f.r.p.f.IndexingBase - build index online index_name="src_index" index_version="4" indexer_id="6ed6524d-01d1-44dc-86fe-1f90face04ff" indexing_method="by records" records_scanned="0" result="failure" should_mark_readable="true"
```

there looks to be a 5 minute gap between starting the index and the failure.  Should this really, even conceivably, be taking more than 5 minutes?

Reply:
To the best of my understanding, it shouldn't be even close to 5 minutes (it's only 1328 records). Something is either hanged or deadlocked, and it never happened on my laptop.